oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/KMeansEvaluation.java

```diff
@@ -160,4 +207,76 @@
         });
   }
 
-}
+  private JavaPairRDD<Integer, Iterable<double[]>> fetchClusteredPoints(JavaRDD<Vector> evalData) {
+
+    return evalData.mapToPair(new PairFunction<Vector, Integer, double[]>() {
+      @Override
+      public Tuple2<Integer, double[]> call(Vector vector) {
+        double closestDist = Double.POSITIVE_INFINITY;
+        int minCluster = -1;
+        double[] vec = vector.toArray();
+
+        for (int i = 0; i < numClusters; i++) {
+          ClusterInfo cluster = clusters.get(i);
+          double distance = distanceFn.distance(cluster.getCenter(), vec);
+          if (distance < closestDist) {
+            closestDist = distance;
+            minCluster = i;
+          }
+        }
+        Preconditions.checkState(minCluster >= 0);
+        return new Tuple2<>(minCluster, vec);
+      }
+    }).groupByKey();
+  }
+
+  private double clusterDissimilarityForPoint(double[] vector, Iterable<double[]> points,
+                                              boolean ownCluster) {
+    double dissimilarity = 0.0;
+    for (double[] point : points) {
+      if (ownCluster && Arrays.equals(vector, point)) {
```

Comment:
This is small, but I think I would just invert the condition, and put the single line of the loop inside it then. Maybe a comment about what the condition checks.